### PR TITLE
BLE: add option to print RSSI on received message

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
KRKNWK-18493
The RSSI it tied to the connection, and it is taken from soft device. Disabling the averaging of the RSSI may increase power consumption see sdc_hci_cmd_vs_set_power_control_request_params() for side effects.

### CODE REVIEW

File: subsys/sal/sid_pal/src/sid_ble_ama_service.c [hunk 1]
```diff
@@ -47,7 +53,25 @@ static ssize_t ama_srv_on_write(struct bt_conn *conn, const struct bt_gatt_attr
 	ARG_UNUSED(conn);
 	ARG_UNUSED(offset);
 	ARG_UNUSED(flags);
+#if CONFIG_SIDEWALK_BLE_RSSI_READ_ON_MESSAGE
+#if CONFIG_SIDEWALK_BLE_RSSI_AVERAGE_DISABLE
+	// avg[n] = gamma * avg[n - 1] + (1 - gamma) * rssi[n]
+	// Here, gamma equals beta/4096, and rssi[n] equals the current RSSI.
+	// leave the last rssi in the average.
```
Comment (ktaborowski) [inline subsys/sal/sid_pal/src/sid_ble_ama_service.c:60]: how is this comment related with code?

File: subsys/sal/sid_pal/src/sid_ble_ama_service.c [hunk 2]
```diff
@@ -47,7 +53,25 @@ static ssize_t ama_srv_on_write(struct bt_conn *conn, const struct bt_gatt_attr
...
+	const sdc_hci_cmd_vs_set_power_control_request_params_t parameters_for_rssi_calculation = {
+		.beta = 4096
+	};
+	sdc_hci_cmd_vs_set_power_control_request_params(&parameters_for_rssi_calculation);
+#endif // CONFIG_SIDEWALK_BLE_RSSI_AVERAGE_DISABLE
+
+	uint16_t conn_handle = 0;
+	bt_hci_get_conn_handle(conn, &conn_handle);
+	const sdc_hci_cmd_vs_read_average_rssi_t p_params = { .conn_handle = conn_handle };
```
Comment (ktaborowski) [inline subsys/sal/sid_pal/src/sid_ble_ama_service.c:69]: why is it `const`?

File: Kconfig.dependencies [hunk 3]
```diff
@@ -279,4 +279,15 @@ config SIDEWALK_DFU_SERVICE_BLE_DEFAULTS
 config MCUMGR_TRANSPORT_BT_AUTHEN
 	default n
 
+config SIDEWALK_BLE_RSSI_READ_ON_MESSAGE
+	bool "Read RSSI on received message"
+	help
+	  Add log message with the current average value of RSSI on every received BLE message
+
+config SIDEWALK_BLE_RSSI_AVERAGE_DISABLE
```
Comment (ktaborowski) [inline Kconfig.dependencies:287]: Isn't It sounds like something configurable on softdevice controller level. Isn't this config redundant to some NCS BLE SDC config?

**Comment by RobertGalatNordic** [comment]
This PR was just an example how to get RSSI value for the connection in NCS. The Current API does not expect this value, so there is no need to keep this change in the source code.
